pkg/http/instrument.go

```diff
@@ -0,0 +1,50 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var (
+	requestDuration = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Name: "http_request_duration_seconds",
+			Help: "Tracks the latencies for HTTP requests.",
+		},
+		[]string{"code", "handler", "method"},
+	)
+
+	requestSize = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name: "http_request_size_bytes",
+			Help: "Tracks the size of HTTP requests.",
+		},
+		[]string{"code", "handler", "method"},
+	)
+
+	requestsTotal = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "http_requests_total",
+			Help: "Tracks the number of HTTP requests.",
+		}, []string{"code", "handler", "method"},
+	)
+)
+
+func init() {
+	prometheus.MustRegister(requestDuration, requestSize, requestsTotal)
```

Comment:
In a follow up we should move this out of an `init` func, in general `init`s should be avoided IMO, since they don't always play nice. Also, we should avoid using the default prometheus registry in favor of an explicitly created one.

Reply:
yes, that is a very good point, maybe even in the very next follow-up.